packages/server-core/src/media/storageprovider/s3.storage.ts

```diff
@@ -303,14 +310,28 @@ export class S3Provider implements StorageProviderInterface {
 
     const args = params.isDirectory
       ? {
-          ACL: ObjectCannedACL.public_read,
+          ACL:
+            projectRegex.test(key) &&
+            !projectPublicRegex.test(key) &&
+            !assetsRegex.test(key) &&
+            !rootImageRegex.test(key) &&
+            !rootSceneJsonRegex.test(key)
+              ? ObjectCannedACL.private
+              : ObjectCannedACL.public_read,
           Body: Buffer.alloc(0),
           Bucket: this.bucket,
           ContentType: 'application/x-empty',
           Key: key + '/'
         }
       : {
-          ACL: ObjectCannedACL.public_read,
+          ACL:
+            projectRegex.test(key) &&
```

Comment:
can we make this a module level function to reuse across the few instances we already do, to avoid discrepancies & regressions in the future

Reply:
This originates in #9345, this PR is based on that since I wanted to re-use some of the regexes I'd made in that one.